### PR TITLE
feat: new attribution github workflow

### PR DESCRIPTION
**Pull request by cryptodev-2s**
## **Description**

This pull request introduces a GitHub Action workflow dedicated to managing attributions effectively within our project. The key components of this new workflow include:

- Local Scripts for Attributions Management:
   - `yarn build:attribution`: Updates attributions to reflect current dependencies.
   - `yarn test:attribution-check`: Checks if attributions are up-to-date with the latest changes.
- GitHub Actions:
   - A check to ensure attributions are up-to-date on all pull requests.
   - An action that allows team members to update attributions directly by commenting `@metamskbot update-attributions` on any PR.

## **Related issues**

Fixes: [#1550](https://github.com/MetaMask/mobile-planning/issues/1550)

### ***Locally***

- Run `yarn build:attribution` to update the attributions file.
- Run `yarn test:attribution-check` to ensure all attributions are current.
- 
### ***On GitHub (CI/PR)***

- Open and push changes to a PR.
- Intentionally remove an entry from attribution.txt to simulate an error.
- Observe that CI fails due to missing attributions.
- Comment `@metmaskbot update-attributions` on the PR.
- A bot will respond with a thumbs up, followed by a comment indicating whether the attributions were successfully updated.

For a practical demonstration, refer to the pull request in my fork: [PR Demonstration](https://github.com/cryptodev-2s/metamask-mobile/pull/4)
### ***Note***
This functionality is currently operational only on a fork as it depends on the integration of the bot action into the develop branch.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .github/workflows/check-attributions.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Check Attributions
+
+on:
+  push:
+    branches: v*
+  pull_request:
+    branches: v*
```
Comment (mcmire) [inline .github/workflows/check-attributions.yml:7]: Is this intended to run on release branches? If so, should this be:

``` suggestion
    branches: release/*
  pull_request:
    branches: release/*

**Review by mcmire — COMMENTED** [review]
Makes sense to me, but I had one question.

File: .github/workflows/check-attributions.yml (hunk 1 above)
Comment (cryptodev-2s) [reply .github/workflows/check-attributions.yml:7]: yes indeed thats what was specified in the ticket

**Review by mcmire — APPROVED** [review]
Looks good to me.

**Pull request merged**